htdocs/core/lib/process_template.lib.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+// Just for display errors in editor
+ini_set('display_errors', 1);
+
+if (!defined('NOTOKENRENEWAL')) {
+	define('NOTOKENRENEWAL', '1'); // Disables token renewal
+}
+if (!defined('NOREQUIREMENU')) {
+	define('NOREQUIREMENU', '1');
+}
+if (!defined('NOREQUIREHTML')) {
+	define('NOREQUIREHTML', '1');
+}
+if (!defined('NOREQUIREAJAX')) {
+	define('NOREQUIREAJAX', '1');
+}
+if (!defined('NOREQUIRESOC')) {
+	define('NOREQUIRESOC', '1');
+}
+require_once '../../main.inc.php';
+require_once './files.lib.php';
+
+top_httphead();
+
+if ($_SERVER['REQUEST_METHOD'] == 'POST' && GETPOSTISSET('content')) {
+	$content = $_POST['content'];
+
+
+	$content = str_replace('<!-- PHP_START -->', '<?php ', $content);
+	$content = str_replace('<!-- PHP_END -->', ' ?>', $content);
+
+	$directory = DOL_DATA_ROOT . '/mailing/email_template';
+	if (!is_dir($directory)) {
+		dol_mkdir($directory);
+	}
+
+	$i = 0;
+	do {
+		$filePath = $directory . '/template_' . $i++ . '.php';
```

Comment:
The $directory = DOL_DATA_ROOT . '/mailing/email_template'; 
is inside the root where binary files are stored. It means the directory should be in readonly mode and is not writable by the web server on a secured installation.
Any temporaty files must be stored into a temp directory.
For example, you can use  $conf->admin->dir_temp

Reply:
ok it s solved